app/src/main/java/com/android/wildex/ui/post/PostDetailsScreen.kt

```diff
@@ -95,18 +141,245 @@ fun PostDetailsScreen(
+                  tint = MaterialTheme.colorScheme.tertiary)
+
+              Text(
+                  text = location,
+                  color = MaterialTheme.colorScheme.tertiary,
+              )
+            }
+
+        Column(
+            modifier = Modifier.weight(1f).padding(horizontal = 8.dp),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              IconButton(
+                  onClick = {
+                    if (likedByCurrentUser) postDetailsScreenViewModel.addLike()
+                    else postDetailsScreenViewModel.removeLike()
+                  }) {
+                    Icon(
+                        imageVector =
+                            if (likedByCurrentUser) Icons.Filled.Favorite
+                            else Icons.Outlined.FavoriteBorder,
+                        contentDescription = if (likedByCurrentUser) "Unlike" else "Like",
+                        modifier = Modifier.size(40.dp),
+                        tint =
+                            if (likedByCurrentUser) MaterialTheme.colorScheme.tertiary
+                            else MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f))
+                  }
+
+              Text(text = likesCount.toString(), color = MaterialTheme.colorScheme.tertiary)
+            }
+      }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CommentSection(
+    modifier: Modifier = Modifier,
+    commentsUI: List<CommentWithAuthorUI> = emptyList(),
+    userId: Id = "",
+    userProfilePictureURL: URL = "",
+    onProfile: (Id) -> Unit = {},
+    postDetailsScreenViewModel: PostDetailsScreenViewModel
+) {
+  Box(
+      modifier =
+          Modifier.padding(8.dp)
+              .border(2.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(12.dp))
+              .clip(RoundedCornerShape(24.dp))) {
+        Column(modifier = modifier.fillMaxWidth()) {
+          LazyColumn(
+              modifier = Modifier.weight(1f).fillMaxWidth(),
+              verticalArrangement = Arrangement.spacedBy(0.dp)) {
+                items(commentsUI) { commentUI ->
+                  Comment(commentUI = commentUI, onProfile = onProfile)
+                }
+              }
+
+          CommentInput(
+              userId = userId,
+              userProfilePictureURL = userProfilePictureURL,
+              onProfile = onProfile,
+              postDetailsScreenViewModel = postDetailsScreenViewModel)
+        }
+      }
 }
+
+@Composable
+fun Comment(
+    commentUI: CommentWithAuthorUI,
+    onProfile: (Id) -> Unit = {},
+) {
+  Box(modifier = Modifier.border(1.dp, MaterialTheme.colorScheme.primary).padding(14.dp)) {
+    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.Top) {
+      ClickableProfilePicture(
+          modifier =
+              Modifier.size(48.dp)
+                  .testTag(ProfileScreenTestTags.COMMENT_PROFILE_PICTURE + commentUI.authorId),
```

Comment:
Maybe consider writing a dedicated function that produces the test tag for a profile picture, as it is done in `NavigationTestTags.kt` in the navigation PR or in the bootcamp for editing todos.
```suggestion
                  .testTag(ProfileScreenTestTags.testTagForProfilePicture(commentUI.authorId),
```